### PR TITLE
Gh15277 campaign price callout

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/15277

### CODE REVIEW

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java [hunk 1]
```diff
@@ -70,10 +77,47 @@ public void onBPartnerChanged(final I_C_Campaign_Price record)
...
 		}
-
+		updatePricingInfo(record);
 		record.setC_BP_Group_ID(-1);
 	}
 
+	private static void updatePricingInfo(final I_C_Campaign_Price record)
+	{
+		final BPartnerId bpartnerId = BPartnerId.ofRepoIdOrNull(record.getC_BPartner_ID());
+		if (bpartnerId == null)
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java:87]: why loading BPartnerId and not checking directly record.getC_BPartner_ID()<=0 then return

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java [hunk 2]
```diff
@@ -70,10 +77,47 @@ public void onBPartnerChanged(final I_C_Campaign_Price record)
...
+		{
+			return;
+		}
+		final IBPartnerDAO bpartnerRepo = Services.get(IBPartnerDAO.class);
+		final CountryId countryId = bpartnerRepo.getDefaultShipToLocationCountryIdOrNull(bpartnerId);
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java:92]: any service must be out of mehtod

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java [hunk 3]
```diff
@@ -70,10 +77,47 @@ public void onBPartnerChanged(final I_C_Campaign_Price record)
...
+		{
+			record.setM_PricingSystem_ID(pricingSystemId.getRepoId());
+			final IPriceListDAO priceListDAO = new PlainPriceListDAO();
+			final PriceListId priceListId = priceListDAO.retrievePriceListIdByPricingSyst(pricingSystemId, countryId, SOTrx.SALES);
+			if (priceListId != null)
+			{
+				final I_M_PriceList_Version newestPriceListVersion = priceListDAO.retrieveNewestPriceListVersion(priceListId);
+				final ProductId productId = ProductId.ofRepoIdOrNull(record.getM_Product_ID());
+				final I_M_ProductPrice productPrice = ProductPrices.retrieveMainProductPriceOrNull(newestPriceListVersion, productId);
+				if (productPrice != null)
```
Comment (cristinamghita) [inline backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java:109]: you shall not retrieve this manually; the price shall be computed using PrincingRUle and from PricingResult you shall take the needed infos.

**Review by cristinamghita — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java (hunk 1 above)
Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java]: correction done

File: backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java (hunk 3 above)
Comment (riadmen) [reply backend/de.metas.business/src/main/java/de/metas/pricing/rules/campaign_price/callout/C_Campaign_Price.java]: correction done

**Review by cristinamghita — APPROVED** [review]
lgtm!

**Pull request merged**